runtime/near-vm-runner/src/wasmer2_runner.rs

```diff
@@ -524,19 +528,6 @@ impl crate::runner::VM for Wasmer2VM {
         )
         .entered();
 
-        if method_name.is_empty() {
-            let error = VMError::FunctionCallError(FunctionCallError::MethodResolveError(
-                MethodResolveError::MethodEmptyName,
-            ));
-            return VMResult::nop_outcome(error);
-        }
-        let artifact =
-            cache::wasmer2_cache::compile_module_cached_wasmer2(code, &self.config, cache);
-        let artifact = match into_vm_result(artifact) {
-            Ok(it) => it,
-            Err(err) => return VMResult::nop_outcome(err),
-        };
```

Comment:
Hm, so this case is actually interesting I think. It's not clear to me if we should charge contract loading cost if compilation error happens, as we don't actually load the contract in this case. 

Hypothetically, if, at some point, we add shape-dependent loading costs (based, eg, on the number of fns in the contract), than we'll have to charge this cost *after* compilation. This seems to be cleaner in some way. 

OTOH, the current strategy of just always charging this cost seems *much* simpler. And, even with shape-dependent costs *and* JIT compilation we might want to charge flat cost before we start compiling, and than shaped cost after we parsed WASM.

So, I guess the current approach is a good one, but, still, an interesting find I didn't realise before!

Reply:
Yeah, we don't necessarily have to charge the loading fee in case of a compilation error. But I to keep it simple until having an actual good reason to make it more complex.

> even with shape-dependent costs and JIT compilation we might want to charge flat cost before we start compiling

Completely agreed. This is the main justification for me why doing it this way really doesn't seem to hurt.

Reply:
Given that we can't be sure if a `FunctionCall` will result in a compilation or not, and we AFAIU don't want conditionally or unconditionally charge the possibly very significant fees for every function invocation, it makes complete sense to me to just charge the “executable loading” fee before compilation.